qiita_db/test/test_sql.py

```diff
@@ -27,5 +35,91 @@ def test_collection_job_trigger(self):
+
+    def test_find_artifact_roots_is_child(self):
+        """Correctly returns the root if the artifact is a child"""
+        sql = "SELECT * FROM qiita.find_artifact_roots(%s)"
+        obs = self.conn_handler.execute_fetchall(sql, [4])
+        exp = [[1]]
+        self.assertEqual(obs, exp)
+
+    def test_find_artifact_roots_is_child_multiple_parents_one_root(self):
+        """Correctly returns the roots if the children has multiple parents
+           but a single root
+        """
+        sql = "SELECT * FROM qiita.find_artifact_roots(%s)"
+        fd, fp = mkstemp(suffix='_table.biom')
+        close(fd)
+        with open(fp, 'w') as f:
+            f.write("test")
+        fp = [(fp, 7)]
+        new = Artifact.create(fp, "BIOM", parents=[Artifact(2), Artifact(3)],
+                              processing_parameters=Parameters(1, Command(1)),
+                              can_be_submitted_to_ebi=True,
+                              can_be_submitted_to_vamps=True)
+        obs = self.conn_handler.execute_fetchall(sql, [new.id])
+        exp = [[1]]
+        self.assertEqual(obs, exp)
+
+    def _create_root_artifact(self):
+        """Creates a new root artifact"""
+        metadata = pd.DataFrame.from_dict(
+            {'SKB8.640193': {'center_name': 'ANL',
+                             'primer': 'GTGCCAGCMGCCGCGGTAA',
+                             'barcode': 'GTCCGCAAGTTA',
+                             'run_prefix': "s_G1_L001_sequences",
+                             'platform': 'ILLUMINA',
+                             'instrument_model': 'Illumina MiSeq',
+                             'library_construction_protocol': 'AAAA',
+                             'experiment_design_description': 'BBBB'}},
+            orient='index')
+        pt = PrepTemplate.create(metadata, Study(1), "16S")
+        fd, fp = mkstemp(suffix='_seqs.fastq')
+        close(fd)
+        with open(fp, 'w') as f:
+            f.write("test")
+        fp = [(fp, 1)]
+        new_root = Artifact.create(fp, "FASTQ", prep_template=pt)
+        return new_root
+
+    def test_find_artifact_roots_is_root_without_children(self):
+        """Correctly returns the root if the artifact is already the root
+           and doesn't have any children
+         """
+        sql = "SELECT * FROM qiita.find_artifact_roots(%s)"
+
+        # Add a new root
+        new_root = self._create_root_artifact()
+        obs = self.conn_handler.execute_fetchall(sql, [new_root.id])
+        exp = [[new_root.id]]
+        self.assertEqual(obs, exp)
+
+    def test_find_artifact_roots_is_child_multiple_parents_multiple_root(self):
+        """Correctly returns the roots if the children has multiple roots"""
+        sql = "SELECT * FROM qiita.find_artifact_roots(%s)"
+
+        new_root = self._create_root_artifact()
+
+        # Add a child of 2 roots
+        fd, fp = mkstemp(suffix='_seqs.fna')
+        close(fd)
+        with open(fp, 'w') as f:
```

Comment:
Should these files be deleted at some point? Or are these being cleaned up after they are moved to their final destination?